src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -2710,6 +2710,190 @@ public static void SoapEncodedSerializationTest_Enum_With_SoapEnumOverrides()
+        XmlTypeMapping myTypeMapping = new SoapReflectionImporter(soapAttributeOverrides).ImportTypeMapping(typeof(SoapEncodedTestType5));
+        var ser = new XmlSerializer(myTypeMapping);
+        var value = new SoapEncodedTestType5()
+        {
+            Name = "MyName",
+            PostitiveInt = "10",
+            Today = new DateTime(2012, 10, 10)
+        };
+
+        var actual = SerializeAndDeserialize(
+            value: value,
+            baseline: "<?xml version=\"1.0\"?>\r\n<SoapEncodedTestType5 xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" id=\"id1\" d1p1:Name=\"MyName\" CreationDate=\"2017-01-27\" xmlns:d1p1=\"http://www.cpandl.com\">\r\n  <PosInt xsi:type=\"xsd:nonNegativeInteger\">10</PosInt>\r\n</SoapEncodedTestType5>",
+            serializerFactory: () => ser);
+
+        Assert.Equal(value.Name, actual.Name);
+        Assert.Equal(value.PostitiveInt, actual.PostitiveInt);
+        Assert.Equal(value.Today, actual.Today);
+    }
+
+    [Fact]
+    [ActiveIssue(15523)]
+    public static void SoapEncodedSerialization_IncludeType()
+    {
+        var soapImporter = new SoapReflectionImporter();
+        soapImporter.IncludeType(typeof(MySpecialOrder));
+        XmlTypeMapping myTypeMapping = soapImporter.ImportTypeMapping(typeof(MyOrder));
+        var ser = new XmlSerializer(myTypeMapping);
+        var value = new MySpecialOrder()
+        {
+            ID = 10,
+            Name = "MyName",
+            SecondaryID = 1000
+        };
+
+        var actual = SerializeAndDeserialize(
+            value: value,
+            baseline: "<?xml version=\"1.0\"?>\r\n<MyOrder xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" id=\"id1\" xsi:type=\"MySpecialOrder\">\r\n  <ID xsi:type=\"xsd:int\">10</ID>\r\n  <Name xsi:type=\"xsd:string\">MyName</Name>\r\n  <SecondaryID xsi:type=\"xsd:int\">1000</SecondaryID>\r\n</MyOrder>",
+            serializerFactory: () => ser);
+
+        Assert.NotNull(actual);
+        Assert.Equal(value.ID, actual.ID);
+        Assert.Equal(value.Name, actual.Name);
+        Assert.Equal(value.SecondaryID, actual.SecondaryID);
+    }
+
+    [Fact]
+    public static void SoapEncodedSerialization_CircularLink()
+    {
+        XmlTypeMapping myTypeMapping = new SoapReflectionImporter().ImportTypeMapping(typeof(MyCircularLink));
+        XmlSerializer ser = new XmlSerializer(myTypeMapping);
+        MyCircularLink value = new MyCircularLink(true);
+        var ms = new MemoryStream();
+        using (var writer = new XmlTextWriter(ms, Encoding.UTF8))
+        {
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartElement("wrapper");
+            ser.Serialize(writer, value);
+            writer.WriteEndElement();
+            ms.Position = 0;
+            string actualOutput = new StreamReader(ms).ReadToEnd();
```

Comment:
You should call `writer.Flush();` before reading the `actualOutput`. You would need to update the baseline below as well, which currently does not contain close element.